### PR TITLE
Features app

### PR DESCRIPTION
**Pull request by rachaouldbabaali**
Ruby on Rails capstone project - Budget app

[Live Demo](https://www.loom.com/share/5d6226804db24e67827578a88a019cc5?sid=12c50f65-acbc-440a-93ef-7353e9092eff)

In this project, I have implemented the following;

- [x] Use ruby gems as software packages system.

- [x] Install Ruby on Rails framework.
- [x] Understand Rails RESTful design and router.
- [x] Use controllers to handle requests and render empty views.
- [x] Use params from browser request in a safe way.
- [x] Use a preprocessed html file with embedded Ruby code.
- [x] Use layouts and templates for shared content.
- [x] Use database migration files to maintain database schema.
- [x] Use validations for models.
- [x] Secure app from n+1 problems.
- [x] Understand what ORM is.
- [x] Write SQL queries with ActiveRecord.
- [x] Set up associations between models.
- [x] Build a webapp that requires the user to log in.
- [x] Use devise gem for authentication.
- [x] Limit access to webapp resources based on authorization rules.
- [x] Analyze in writing why you have made a coding choice using one structure over another

### CODE REVIEW

File: spec/features/expenses_spec.rb [hunk 1]
```diff
@@ -0,0 +1,39 @@
+require 'rails_helper'
+
+RSpec.describe 'Expenses', type: :feature do
+  before(:each) do
+    @user = User.create(name: 'Racha Oba', email: 'racha@gmail.com', password: '123456',
+                        password_confirmation: '123456', confirmation_token: nil, confirmed_at: Time.now)
+    @group = Group.create(name: 'Food', icon: 'icons/food.PNG', user_id: @user.id)
```
Comment (Davidosky007) [inline spec/features/expenses_spec.rb]: * I get errors when I try to run your tests seems the errors come from the icons in your test as they are different from what you have in your assets folder Kindly put in the correct image title and extension and it should work for example( 'icons/food.jpg') 
image

File: README.md [hunk 2]
```diff
@@ -1 +1,159 @@
...
+<!-- Features -->
+
+### Key Features <a name="key-features"></a>
+
+- **[Users can view their transaction and Categories]**
+- **[Users can create a transaction]**
+- **[Users can create a category.]**
+- **[Limit access to web app resources based on authorization rules.]**
+
+- Live Demo Link: will be updated when available
```
Comment (Davidosky007) [inline README.md]: * Kindly add a working live demo link as it's a [requirement ](https://github.com/microverseinc/curriculum-rails/blob/main/capstone/rails_capstone.md#:~:text=The%20project%20should%20be%20deployed%20and%20accessible%20online.)

File: app/assets/stylesheets/style.css [hunk 3]
```diff
@@ -0,0 +1,230 @@
...
+  height: 55px;
+  padding-top: 12px;
+}
+
+p.amount {
+  font-weight: bold;
+  padding: 8px;
+}
+
+.total_amount {
```
Comment (Davidosky007) [inline app/assets/stylesheets/style.css:54]: * As required : 
* You should follow these [design guidelines](https://www.behance.net/gallery/19759151/Snapscan-iOs-design-and-branding?tracking_source=), including:
Colors.
Typography: font face, size and weight.
Layout: composition and space between elements.
* Kindly use the given fonts and colors
image
image

**Review by Davidosky007 — CHANGES_REQUESTED** [review]
Hi @rachaouldbabaali 👋, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!
## Highlights

* N+1 issue solved 👍 
* Well-written PR 👍 
* No linter errors 👍 
* Video presentation looks good 👍 

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better._
Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me @davidsoky007 in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: README.md [hunk 4]
```diff
@@ -1 +1,160 @@
...
+<!-- Features -->
+
+### Key Features <a name="key-features"></a>
+
+- **[Users can view their transaction and Categories]**
+- **[Users can create a transaction]**
+- **[Users can create a category.]**
+- **[Limit access to web app resources based on authorization rules.]**
+
+- Live Demo [Link](https://budget-app-racha.onrender.com/)
```
Comment (DammyShittu) [inline README.md:44]: - [ ] Kindly ensure your app is well deployed. It shows an error when I click the link.

  image

**Review by DammyShittu — CHANGES_REQUESTED** [review]
Hi @rachaouldbabaali,

## Highlights
✔️ Good styling and design for the app
✔️ Passing tests

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by rachaouldbabaali** [comment]
thank you @DammyShittu the issue is fixed

**Review by descholar-ceo — APPROVED** [review]
Hi @rachaouldbabaali 🤝 , 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## Highlights
👍🏽 Your project passes all test cases ✔️ 
👍🏽 Now The project's live demo link works fine ✔️ 
👍🏽 The styling is commendable ✔️ 
👍🏽 The readme file is professional and it has all headers / sections as described from microverse's readme template ✔️
👍🏽 The video recording explains the project clearly ✔️ 

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by rachaouldbabaali** [comment]
Thank you , I truly appreciate your time and efforts in reviewing my work

**Pull request merged**